pkg/cmd/cli/nodeagent/server.go

```diff
@@ -333,6 +337,62 @@ func (s *nodeAgentServer) markInProgressCRsFailed() {
 	s.markInProgressPVRsFailed(client)
 }
 
+func (s *nodeAgentServer) markDataUploadsCancel(client ctrlclient.Client, r *controller.DataUploadReconciler) {
+	pods := &v1.PodList{}
+	if err := client.List(s.ctx, pods); err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to list pods on current node")
+		return
+	}
+	for _, pod := range pods.Items {
+		du, err := r.FindDataUploadByPod(&pod)
+		if err != nil {
+			s.logger.WithError(errors.WithStack(err)).Errorf("failed to get dataupload by pod %s/%s", pod.Namespace, pod.Name)
+			continue
+		}
+		updated := du.DeepCopy()
+		updated.Spec.Cancel = true
+		du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceled
+		if du.Status.StartTimestamp.IsZero() {
+			du.Status.StartTimestamp = &metav1.Time{Time: r.Clock.Now()}
+		}
+		du.Status.CompletionTimestamp = &metav1.Time{Time: r.Clock.Now()}
+		updated.Status.Message = fmt.Sprintf("found a dataupload with status %q during the server starting, mark it as cancel", du.Status.Phase)
```

Comment:
Change it to `during node-agent starting`? Otherwise, the messages for both velero server restarting and node-agent restarting are the same, we will not be able to tell what exactly happened during troubleshooting

Reply:
Modified